packages/core/src/components/Modal/ModalHeader/ModalHeader.tsx

```diff
@@ -70,6 +62,18 @@ export interface ModalHeaderProps extends VibeComponentProps {
   hideCloseButton?: boolean;
 }
 
+interface ModalHeaderWithOnlyTitle extends BaseModalHeaderProps {
+  title: ElementContent;
+  children?: never;
+}
+
+interface ModalHeaderWithOnlyChildren extends BaseModalHeaderProps {
+  title?: never;
+  children: ElementContent;
+}
```

Comment:
For some reason the `title` prop was mandatory even when `children` were provided, it came up in the story so I changed it. Now either of them is mandatory